File: app/src/main/java/com/example/pam_app/model/Income.java
package com.example.pam_app.model;

import java.io.Serializable;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public class Income implements Serializable {
    public final Integer id;
    private final String title;
    private final double amount;
    private final IncomeType incomeType;
    private final Date date;

    public Income(final String title, final double amount, final IncomeType incomeType, final Date date) {
        this.title = title;
        this.amount = amount;
        this.incomeType = incomeType;
        this.date = date;
        this.id = null;
    }

    public Income(final int id, final String title, final double amount, final IncomeType incomeType, final Date date) {
        this.title = title;
        this.amount = amount;
        this.incomeType = incomeType;
        this.date = date;
        this.id = id;
    }

    public Integer getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public double getAmount() {
        return amount;
    }

    public String getAmountString() {
        return "$" + amount;
    }

    public String getDateString() {
        DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd");
        return dateFormat.format(date);
    }

    public IncomeType getIncomeType() {
        return incomeType;
    }

    public Date getDate() {
        return date;
    }
}

File: app/src/main/java/com/example/pam_app/model/Income.kt
package com.example.pam_app.model

import android.annotation.SuppressLint
import java.io.Serializable
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.*

class Income : Serializable, Entry {
    val id: Int?
    override val comment: String?
    val amount: Double
    val incomeType: IncomeType
    val date: Date?

    constructor(title: String?, amount: Double, incomeType: IncomeType, date: Date?) {
        comment = title
        this.amount = amount
        this.incomeType = incomeType
        this.date = date
        id = null
    }

    constructor(id: Int, title: String?, amount: Double, incomeType: IncomeType, date: Date?) {
        comment = title
        this.amount = amount
        this.incomeType = incomeType
        this.date = date
        this.id = id
    }

    override val amountString: String
        get() = "$$amount"

    override val dateString: String
        @SuppressLint("SimpleDateFormat")
        get() {
            val dateFormat: DateFormat = SimpleDateFormat("yyyy/MM/dd")
            @Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
            return dateFormat.format(date)
        }

}